### ISSUE TITLE
ArgumentException / ArgumentNullException when checking strings

### ISSUE THREAD

**Issue by DominikAmon** [mattwhitfield/Unitverse#257]
When generating unit tests for the following code:

```
public static class Sample
{
	public static void SampleMethod(string input)
	{
		ArgumentException.ThrowIfNullOrWhiteSpace(input);
	}
}
```

the following check will be created:

```
[DataTestMethod]
[DataRow(null)]
[DataRow("")]
[DataRow("   ")]
public void CannotCallSampleMethodWithInvalidInput(string value)
{
   Assert.ThrowsException<ArgumentNullException>(() => Sample.SampleMethod(value));
}
```

When using [ArgumentException.ThrowIfNullOrWhiteSpace](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception.throwifnullorwhitespace) the behavior works as such:

* `ArgumentNullException`: argument is `null`.
* `ArgumentException`: argument is empty or consists only of white-space characters.

Suggested generation:
```
[TestMethod]
public void CannotCallSampleMethodWithNullInput()
{
	Assert.ThrowsException<ArgumentNullException>(() => Sample.SampleMethod(null));
}

[DataTestMethod]
[DataRow("")]
[DataRow("   ")]
public void CannotCallSampleMethodWithInvalidInput(string value)
{
	Assert.ThrowsException<ArgumentException>(() => Sample.SampleMethod(value));
}
```

**Comment by mattwhitfield** [comment]
I went to implement this - and it actually requires a bit more effort than it would seem - because I don't currently have the facility to use net8.0 specific methods in the integration tests. It's something that will need addressing at some point. If I can think of a quick way around it then I will try and get it in

**Comment by mattwhitfield** [comment]
I added support for this - it will emit in the mode you suggested when either:
a) `ArgumentException.ThrowIfNullOrWhiteSpace` is used in the method body
or
b) The option `UseSeparateChecksForNullAndEmpty` is set to `true`

The detection is also pretty basic. So - for a method with two string parameters it won't try and check if `ArgumentException.ThrowIfNullOrWhiteSpace` is used for the specific parameter that the test is being generated for - it will just check if it's used in the method body at all. Also, for constructor checks, if multiple constructors have a parameter with the same name, it will generate split tests if `ArgumentException.ThrowIfNullOrWhiteSpace` is used in any of those constructors. 

I think that's OK from a consistency perspective - but just wanted to be clear about the limits. 😊

It should be available in 0.213...
